runtimes/runtimes/standalone.ts

```diff
@@ -249,7 +251,11 @@ export const standalone = (props: RuntimeProps) => {
                 chat = new BaseChat(lspConnection)
             }
 
-            return s({ chat, credentialsProvider, lsp, workspace, telemetry, logging, runtime })
+            const identityManagement: IdentityManagement = {
+                onListProfiles: handler => lspConnection.onRequest(listProfilesRequestType, handler),
+            }
+
+            return s({ chat, credentialsProvider, lsp, workspace, telemetry, logging, runtime, identityManagement })
```

Comment:
The final interface with parameters that we inject becomes confusing for server, in my opinion. Part of the parameters are basic services that servers just consume, like `credentialsProvider`, `workspace`, `telemetry`, logging` and `runtime`. `lsp` and `chat` are specific, because those are LSP-specific and servers need to implement them. `identityManagement` also belongs to the 2nd group. I think we should not continue defining those LSP-specific handlers that server needs to implement here, but rather define them in `extensions` of `lsp`, as we did for `onInlineCompletionWithReferences`.
Keep in mind that with Toolkits common components, this list will grow further.

Reply:
I don't think IdentityManagement is really part of `Lsp`, as in, the _Language Server Protocol_. It's definitely a runtime feature and capability that servers can implement, with or without doing anything with programming languages.

Reply:
It seems more confusing to me we have a `credentialsProvider` and something that is _effectively_ a credentials provider, but one provides credentials to `Server`s and the other manages credentials for clients.

We used to have an `Auth` feature, is it time to bring that back and merge the two?

Reply:
There is an Auth feature.  

The original proposal in the TKC: Auth Design was to extend that Auth feature to respond to the JSON-RPC requests to handle client identity management functions (such as SSO login).  The Auth feature exposes the CredentialsProvider interface injected into the LSP servers which offers only limited functionality that the servers need.  The client functions such as setting and erasing the current creds/token are only exposed to clients via the JSON-RPC interface, not to LSP servers via the CredentialsProvider interface.

The choice was made to move the client functions into an LSP server instead for clear lines of code ownership and to fit the expected workflow for 3rd-party contributors among other things.  I do get volodkevych@'s point about separate responsibilities though as most of these injected features "support the server" (e.g. I need a secret to call this API, what is it?) whereas the identityManagement feature primarily creates a means to expose non-LSP API calls to LSP servers to handle (similar to what the Lsp server-interface does for LSP).

https://github.com/aws/language-server-runtimes/blob/main/runtimes/protocol/auth.ts
https://github.com/aws/language-server-runtimes/blob/main/runtimes/runtimes/auth/auth.ts
https://github.com/aws/language-server-runtimes/blob/main/runtimes/server-interface/auth.ts

Something like:

```
s({ chat, credentialsProvider, lsp, workspace, telemetry, logging, runtime, extensions: { identityManagement: identityManagement } })
```

It makes sense to avoid further bloating the parameter list and for most servers, they can probably safely ignore the extensions arg entirely.  Though "extensions" is ambiguous.  Maybe a different name such as `apis` or `apiExtensions` would be better fit.

Reply:
We've had some offline discussions about this. To summarize:

There is an agreement that the `identityManagement` feauture doesn't belong to the `lsp`. 
Finding the long-term approach on how we separate `apiExtenstions`/`features` (`what servers use` vs `what servers implement`) is better to do separately from this PR because we don't have an agreement on whether we need to change the interface at all and the proposed ideas are backward incompatible anyway. 

Thus I propose to keep the `identityManagement` inside the `features` object in this PR.